gslib/context_config.py

```diff
@@ -107,6 +113,55 @@ def _SplitPemIntoSections(contents, logger):
   return result
 
 
+def _check_path():
+  """Checks for context aware metadata. If it exists, returns the absolute path;
+  otherwise returns None.
+  Returns:
+      str: absolute path if exists and None otherwise.
+  """
+  metadata_path = os.path.expanduser(_CONTEXT_AWARE_METADATA_PATH)
+  if not os.path.exists(metadata_path):
+    return None
+  return metadata_path
+
+
+def _read_metadata_file(metadata_path):
+  """Loads context aware metadata from the given path.
+  Returns:
+      Dict[str, str]: The metadata.
+  Raises:
+      CertProvisionError: If failed to parse metadata as JSON.
+  """
+  try:
+    with open(metadata_path) as f:
+      metadata = json.load(f)
+  except ValueError as e:
+    raise CertProvisionError(e)
+  return metadata
+
+
+def _default_command():
+  """Loads default cert provider command.
+  Returns:
+      str: The default command.
+  Raises:
+      CertProvisionError: If command cannot be found.
+  """
+  metadata_path = _check_path()
+  if metadata_path:
```

Comment:
I would restructure this as:

```
if not metadata_path:
  raise

metadata_json = _read...
```

That way, you can unnest most of the code, and you can stop reading earlier if you know _check_path returned false. Speaking of which, I would recommend differentiating the two `CertProvisionError` messages a little.